catalog/app/components/JsonEditor/State.js

```diff
@@ -48,50 +45,42 @@ export function validateOnSchema(optSchema) {
   const schema = optSchema || emptySchema
 
   const ajv = new Ajv({ schemaId: 'auto' })
-  const validate = ajv.compile(schema)
 
-  return (obj) => {
-    validate(obj)
-    return validate.errors || []
+  try {
+    const validate = ajv.compile(schema)
+
+    return (obj) => {
+      validate(obj)
+      return validate.errors || []
+    }
+  } catch (e) {
+    return () => [e]
   }
 }
 
-function serializeAddress(addressPath) {
-  return addressPath.join(', ')
-}
+const serializeAddress = (addressPath) => addressPath.join(', ')
 
-function getAddressPath(key, parentPath) {
-  if (key === '') return parentPath
-  return (parentPath || []).concat(key)
-}
+const getAddressPath = (key, parentPath) =>
+  key === '' ? parentPath : (parentPath || []).concat(key)
 
-function getSchemaType(item) {
-  return item.type
-}
+const getSchemaType = R.prop('type')
 
-function getSchemaItem({ item, sortIndex, key, parentPath, required }) {
-  return {
-    address: getAddressPath(key, parentPath),
-    required,
-    valueSchema: item,
-    sortIndex,
-    type: getSchemaType(item),
-  }
-}
+const getSchemaItem = ({ item, sortIndex, key, parentPath, required }) => ({
+  address: getAddressPath(key, parentPath),
+  required,
+  valueSchema: item,
+  sortIndex,
+  type: getSchemaType(item),
+})
 
-function assocObjValue(objPath, value, obj) {
-  return R.assocPath(objPath, value, obj)
-}
+const assocObjValue = R.assocPath
```

Comment:
do we really need this alias?

Reply:
Yes, to have consistently similar functions for an original object and a schema. It's a coincidence, that function's implementation is an alias to `R.assocPath`